cmd/minishift/cmd/start.go

```diff
@@ -154,42 +128,50 @@ func runStart(cmd *cobra.Command, args []string) {
 	}
 
 	//Create the host directories if not present
-	hostDirs := []string{viper.GetString(startFlags.HostConfigDir.Name),
-		viper.GetString(startFlags.HostDataDir.Name), viper.GetString(startFlags.HostVolumeDir.Name),
-		viper.GetString(startFlags.HostPvDir.Name)}
-	err = clusterup.EnsureHostDirectoriesExist(libMachineClient, hostDirs)
-	if err != nil {
-		fmt.Println("Error creating required host directories: ", err)
-		atexit.Exit(1)
+	hostDirs := []string{
+		viper.GetString(startFlags.HostConfigDir.Name),
+		viper.GetString(startFlags.HostDataDir.Name),
+		viper.GetString(startFlags.HostVolumeDir.Name),
+		viper.GetString(startFlags.HostPvDir.Name),
 	}
-
-	ip, err := host.Driver.GetIP()
+	err = clusterup.EnsureHostDirectoriesExist(hostVm, hostDirs)
 	if err != nil {
-		fmt.Println("Error determining host ip: ", err)
-		atexit.Exit(1)
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating required host directories: %v", err))
 	}
 
-	if proxyConfig.IsEnabled() {
-		// once we know the IP, we need to make sure it is not proxied in a proxy environment
-		proxyConfig.AddNoProxy(ip)
-		proxyConfig.ApplyToEnvironment()
+	autoMountHostFolders(hostVm.Driver)
+
+	openShiftVersion := clusterup.DetermineOpenShiftVersion(viper.GetString(startFlags.OpenshiftVersion.Name))
```

Comment:
I am lost here. We are are talking about a single line here, and you want me to make three lines and a function call out of this. AFAICT it also does not increase clarity here. 

> as it stands now, the GetString is likely used to override the value we seed. 

It does not override anything, it just gets it from Viper. The return value from clusterup.DetermineOpenShiftVersion does an override of sort, but I think that is well documented in the Godocs of clusterup.DetermineOpenShiftVersion.